packages/next/server/config.ts

```diff
@@ -539,7 +534,11 @@ function assignDefaults(userConfig: { [key: string]: any }) {
     result.pageExtensions = pageExtensions
   }
 
-  return result
+  return {
+    ...result,
+    images: result.images as NextConfigComplete['images'], // runtime checks doesn't seem to narrow down the type
+    typescript: result.typescript as NextConfigComplete['typescript'], // defaultConfig already has default values
```

Comment:
Hmm, I thought the point of this PR was to solve the incorrect types 🤔

Reply:
These were a bit complex, see the comments. Those are config objects, and the runtime checks did not narrow the types properly. I'll have another look.